### PR TITLE
Allow round trip to work on nxos_vlans

### PR DESCRIPTION
**Pull request by NilashishC**
##### SUMMARY
- Make sure round trip works with nxos_vlans module.
- Pop `mode` key from want/have for platforms that don't support it.
- Allow configuring Vlan 1 through the module.
- Overhaul unit tests

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
nxos_vlans.py

### CODE REVIEW

File: tests/integration/targets/nxos_vlans/tests/common/rtt.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    - name: Revert back to base config using facts round trip
+      register: revert
+      cisco.nxos.nxos_vlans:
+        config: "{{ ansible_facts['network_resources']['vlans'] }}"
+        state: overridden
+
+    - assert:
+        that:
+          - base_config['after'] == revert['after']
```
Comment (justjais) [inline tests/integration/targets/nxos_vlans/tests/common/rtt.yaml:60]: @NilashishC from where the revert value is getting populated?

Comment (NilashishC) [reply tests/integration/targets/nxos_vlans/tests/common/rtt.yaml:60]: From the previous task, in line 53?

Comment (justjais) [reply tests/integration/targets/nxos_vlans/tests/common/rtt.yaml:60]: cool, +1

**Review by justjais — APPROVED** [review]
lgtm

**Pull request merged**